library/Test/Hspec/JUnit/Config/Env.hs

```diff
@@ -0,0 +1,48 @@
+-- | Load a 'JUnitConfig' using environment variables
+module Test.Hspec.JUnit.Config.Env
+    ( envJUnitEnabled
+    , envJUnitConfig
+    ) where
+
+import Prelude
+
+import Data.Semigroup (Endo(..))
+import Data.Text (pack)
+import System.Directory (getCurrentDirectory)
+import System.Environment (lookupEnv)
+import System.FilePath (takeBaseName)
+import Test.Hspec.JUnit.Config
+
+-- | Is @JUNIT_ENABLED=1@ set in the environment?
+envJUnitEnabled :: IO Bool
+envJUnitEnabled = (== Just "1") <$> lookupEnv (envPrefix <> "ENABLED")
+
+-- | Produce a 'JUnitConfig' by reading environment variables
+--
+-- Variable names align with setter functions from "Test.Hspec.JUnit.Config":
+--
+-- * @JUNIT_OUTPUT_DIRECTORY@ 'setJUnitConfigOutputDirectory'
+-- * @JUNIT_OUTPUT_NAME@ 'setJUnitConfigOutputName
+-- * and so on
+--
+envJUnitConfig :: IO JUnitConfig
+envJUnitConfig = do
+  modify <- appEndo . foldMap Endo <$> sequence
+    [ lookupEnvOverride "OUTPUT_DIRECTORY" setJUnitConfigOutputDirectory
+    , lookupEnvOverride "OUTPUT_NAME" setJUnitConfigOutputName
+    , lookupEnvOverride "OUTPUT_FILE" setJUnitConfigOutputFile
+    , lookupEnvOverride "SUITE_NAME" $ setJUnitConfigSuiteName . pack
+    , lookupEnvOverride "SOURCE_PATH_PREFIX" setJUnitConfigSourcePathPrefix
+    ]
```

Comment:
Are these env vars standardized somewhere? Where are they coming from?

Reply:
Stop trying to make JUnit standardization happen.

![](https://i.ytimg.com/vi/Pubd-spHN-0/maxresdefault.jpg)

The original design of `JUnitConfig` was inspired by `jest-junit`'s environment variables. And then these variables are designed to match `JUnitConfig` (as the docs say). jest-junit uses these variables with the prefix `JEST_JUNIT_`, but I elected to use simply `JUNIT_` here. Feel free to push for something else.